Use OS certificate bundle

kbrock opened the pull request:
Part of https://github.com/ManageIQ/manageiq-appliance/pull/341

This gets database ssl keys out of `/root`, which is only accessible to `root` user, and into a standard location which is accessible to other users, namely user `manageiq`

1. Store the key that signed the postgres server in the standard location
2. No longer store `root.crt` in `/root` so user manageiq can use it
3. Still configure postgres for public and private key
4. No longer store `postgres.key` in /var/vmdb/ since `postgres` can not access those files.
5. No longer configure postgres with a root cert that verifies the cert of the connecting client. This is typically `ssl_ca_file=root.crt`. We only use server side certificates and not client side certificates.

kbrock commented:
/cc @jrafanie could you take a peek to see if this works like you thought it should?

Fryguy commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -12,10 +12,11 @@ For more information, see [Generating SSL Certificates for Your Appliance and Da
-
   - `/var/www/miq/vmdb/certs/server.cer.key` - Private key for server certificate.
 
+<!--
```
This PR should just delete this unless it's a WIP PR.

Fryguy replied:
Similar for the other comments

Fryguy commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -83,21 +91,22 @@ The database appliance now accepts only connections from connecting appliances t
+3.  Copy the root certificate file from the database to the operating system's list of valid certificates:
 
-        [root@{{ site.data.product.title_short_l }}2 ~]# mkdir /root/.postgresql
+        [root@{{ site.data.product.title_short_l }}2 ~]# DATABASE=[database_appliance_fqdn]
```
I think this can be simplfied to just show a terminal prompt and doesn't need the product info...

```suggestion
        $ DATABASE=[database_appliance_fqdn]
```

Fryguy replied:
Unless this is meant to contrast between 2 different terminal sessions?  If so, then `root@appliance-host` is just simpler

kbrock replied:
do you want me to make this change across the board?

bdunne commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -83,21 +91,22 @@ The database appliance now accepts only connections from connecting appliances t
 
-        [root@{{ site.data.product.title_short_l }}2 ~]# mkdir /root/.postgresql
+        [root@{{ site.data.product.title_short_l }}2 ~]# DATABASE=[database_appliance_fqdn]
+        [root@{{ site.data.product.title_short_l }}2 ~]# scp root@$DATABASE:/var/www/miq/vmdb/certs/root.crt /etc/pki/ca-trust/source/$DATABASE-postgres
```
I think this needs to be in `/etc/pki/ca-trust/source/anchors/`

bdunne commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -83,21 +91,22 @@ The database appliance now accepts only connections from connecting appliances t
 4.  Test the connection between the connecting appliance and the database appliance by using the `psql`command:
 
-        [root@localhost ~]# psql -h [database_appliance_fqdn] -d vmdb_production
+        [root@localhost ~]# PGSSLMODE=verify-full psql -h [database_appliance_fqdn] -d vmdb_production "dbname=vmdb_production sslmode=verify-full"
```
Is the quoted text supposed to be there?

kbrock replied:
thnx. that is how most suggested it but I put in the `ENV` var to get better success in my environment

kbrock replied:
hmm. translate what I typed

> thanks. made that change.

bdunne commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -106,6 +115,9 @@ The database appliance now accepts only connections from connecting appliances t
     The `psql` displays information about the SSL connection, which indicates that the configuration succeeded. Enter `\q` to leave `psql`.
+    To understand the values for `PGSSLMODE` please see [postgress sslmode documentation].
+
+[postgres ssl documentation](https://www.postgresql.org/docs/9.0/libpq-connect.html#LIBPQ-CONNECT-SSLMODE)
```
Should this use the v12 link? https://www.postgresql.org/docs/12/libpq-connect.html#LIBPQ-CONNECT-SSLMODE

jrafanie commented:
LGTM once the comments are addressed.  👍

kbrock commented:
push: fixed updates

- changed unix prompt in examples
- removed comments remaining for root.crt
- fixed cert target directory to include `anchors`
- fixed postgres documentation link. Also changed the wording a little for that link.
- fixed markdown for postgres documentation link
- added headings for each section
- added check/command to ensure

Fryguy commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -106,6 +93,17 @@ The database appliance now accepts only connections from connecting appliances t
+
+5.  Ensure `/etc/default/manageiq.properties` has the proper ssl key file location defined:
+
+        $ `grep PGSSLROOTCERT /etc/default/manageiq.properties ` || \
```
Super minor, but these backticks within a code block are just weird to me.  Minor, but can we do:

```suggestion
        $ $(grep PGSSLROOTCERT /etc/default/manageiq.properties) || \
```

Fryguy commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -106,6 +93,17 @@ The database appliance now accepts only connections from connecting appliances t
     The `psql` displays information about the SSL connection, which indicates that the configuration succeeded. Enter `\q` to leave `psql`.
+    The `PGSSLMODE` variable forces a certain level of SSL. To learn more, please see [postgress sslmode documentation].
+
+    [postgress sslmode documentation]: https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLMODE
```
If we're gonna use 14, then might as well use current, so we don't have to keep updating.

```suggestion
    [postgress sslmode documentation]: https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNECT-SSLMODE
```

Fryguy commented on appliance_hardening_guide/_topics/db_ssl.md:
```diff
@@ -74,30 +58,33 @@ To configure SSL on the database appliance:
 4.  Test the connection between the connecting appliance and the database appliance by using the `psql`command:
 
-        [root@localhost ~]# psql -h [database_appliance_fqdn] -d vmdb_production
+        $ PGSSLMODE=verify-full psql -h [database_appliance_fqdn] -d vmdb_production
```
I wonder if it's simpler, instead of psql followed by a separate doc line to do \q, to just have a single line something like:

```sh
$ PGSSLMODE=verify-full psql -h [database_appliance_fqdn] -d vmdb_production -c "SELECT * FROM pg_stat_ssl"
```

or perhaps filtered by only the current pid if there is more than one connection:

```sh
$ PGSSLMODE=verify-full psql -h [database_appliance_fqdn] -d vmdb_production -c "SELECT * from pg_stat_ssl WHERE pid=pg_backend_pid()"
```

Fryguy reviewed (approved):
LGTM - @bdunne Can you take another look?
